components/core/CMakeLists.txt

```diff
@@ -47,6 +47,15 @@ if (IS_BIG_ENDIAN)
     message(FATAL_ERROR "Big-endian machines are not supported")
 endif()
 
+# Set log surgeon library
+set(log_surgeon_DIR "/home/sharaf/.local/lib/cmake/log_surgeon/")
+find_package(log_surgeon REQUIRED)
+if(log_surgeon_FOUND)
+    message(STATUS "Found spdlog ${log_surgeon_VERSION}")
+else()
+    message(FATAL_ERROR "Could not find static libraries for log_surgeon")
+endif()
+
```

Comment:
Add log surgeon as a git submodule and use `add_subdirectory` to build/include it.

Reply:
commit 395345a49b349b20951659bb412866c060b152c1